test/box-luatest/downgrade_test.lua

```diff
@@ -973,3 +973,40 @@ g.test_downgrade_global_names = function(cg)
         t.assert_equals(#box.space._cluster:format(), 2)
     end)
 end
+
+-----------------------------
+-- Check downgrade from 3.1.0
+-----------------------------
+
+g.test_downgrade_func_trigger = function(cg)
+    cg.server:exec(function()
+        local helper = require('test.box-luatest.downgrade_helper')
+        local trigger = require('trigger')
+        local _func = box.space._func
+        local prev_version = helper.prev_version(helper.app_version('3.0.0'))
+        local lua_code = 'function(a, b) return a + b end'
+        t.assert_equals(trigger.info('test'), {})
+        box.schema.func.create('trigger_func',
+            {body = lua_code, trigger = 'test'})
+        local errmsg = 'Function trigger_func is registered as ' ..
+            'event trigger. It is supported starting from version 3.1.0'
+        t.assert_equals(box.schema.downgrade_issues(prev_version), {errmsg})
+        -- Check if cannot actually downgrade
+        t.assert_error_msg_content_equals(errmsg, box.schema.downgrade,
+            prev_version)
+        box.schema.func.drop('trigger_func')
+        box.schema.func.create('non_trigger_func',
+            {body = lua_code, trigger = {}})
+        t.assert_equals(box.schema.downgrade_issues(prev_version), {})
+        t.assert_equals(#_func:format(), 20)
+        t.assert_equals(_func:format()[20].name, 'trigger')
+        -- 2 for idempotence.
+        for _ = 1, 2 do
```

Comment:
Nice usage of loop for idempotence check!

Reply:
Thanks :)